### PR TITLE
Fixes to align to DSL

### LINKED ISSUE #100: Button: the ebay-button raw is visually incorrect in iOS
The "Raw" button does not look visually correct. Specifically, the font looks off.

ios-raw-button

The expected appearance is ... TBD.

(Originally reported here: https://github.com/eBay/ebayui-core/issues/60)

### LINKED ISSUE #106: Button: type "small" should change font size to 14px
When the class `btn--small` is applied, only the paddings change. It should also change the font size to `14px`.

**Note:** We need design to confirm this font size before proceeding.

### LINKED ISSUE #127: Button: secondary button does not match DSL
http://ebay.github.io/skin/ds6/#button
https://dsl.ebay.com/ds6/buttons
text for secondary button is black instead of blue

### PR DESCRIPTION
**Pull request by seangates**
## Description
- fix secondary button color, hover state, and active states
- fix for primary button hover state
- fix for small button font size
- webkit-specific fix for the raw button
- separate some styles from the expand button, which does not share the same colors

## References
Closes #127 
Closes #106 
Closes #100

### CODE REVIEW

**Comment by ianmcburnie** [comment]
@seangates Looks like a regression to the CTA button:

screen shot 2018-04-18 at 3 29 59 pm

**Comment by seangates** [comment]
@ianmcburnie Crap. I know what that is. Let me fix it.

**Comment by ianmcburnie** [comment]
Menu buttons appear broken too:

screen shot 2018-04-18 at 4 34 25 pm

**Comment by ianmcburnie** [comment]
Yep, I did.

File: src/less/button/ds6/button-base.less [hunk 1]
```diff
@@ -4,6 +4,11 @@
 // so all font-sizes are defined with px
 
 // base properties for all buttons
+button:not([class*="btn--"]) {
```
Comment (ianmcburnie) [inline src/less/button/ds6/button-base.less]: @seangates Why are we adding a global rule? Which issue does this address? If there is a legitimate need for this it should go in global module.

**Comment by ianmcburnie** [comment]
@seangates Raw button has changed from this:

screen shot 2018-04-18 at 5 44 46 pm

to this:

screen shot 2018-04-18 at 5 44 21 pm

Is that intentional?

**Comment by seangates** [comment]
@ianmcburnie Yes, this is to support iOS safari, which rounds the corners of the button. I don't have a way to target iOS specifically, so I have to opt for changing the way the raw button looks.

Now, I tried the idea of adding a white background, or a different colored background to make them all consistent. But, I ended up opting for whatever background the browser decided to use. In this case, Safari decided on a darker one 🤷‍♂️ ?

File: src/less/button/ds6/button-base.less (hunk 1 above)
Comment (seangates) [reply src/less/button/ds6/button-base.less]: Let me adjust this. You're right that it shouldn't be global

**Comment by seangates** [comment]
@ianmcburnie I scoped the raw button fix to just the buttons with the `.btn` class. This should fix it properly for only those buttons that use the base class.

**Comment by ianmcburnie** [comment]
Ok. Is there any reason these rules can't just go on the base selector instead of using a `:not` selector?

Also I hope teams aren't using the raw button "as is". The idea is to "Use the raw button as the base to build custom modifiers."

**Comment by seangates** [comment]
The other modifiers don't reset the appearance, though. Do we want `-webkit-appearance: button` to be part of the base we build from?

**Comment by ianmcburnie** [comment]
Hmm, but the point of the btn class is that it forms the base for all modifiers to build on (our modifiers and custom modifiers). If we remove some of that base when the modifiers are applied (via the not selector), will that not lead to unpredictable results?

**Comment by seangates** [comment]
Let me try it out.

File: src/less/button/ds6/button-base.less [hunk 2]
```diff
@@ -65,15 +67,11 @@ button.btn--primary,
 a.fake-btn--primary {
     background-color: @ds6-color-p002-blue;
     border: 1px solid @ds6-color-p002-blue;
```
Comment (ianmcburnie) [inline src/less/button/ds6/button-base.less]: Nitpicking now, but now that we have a border type and thickness set in base class, we can just specify the border-color in the modifiers.

File: dist/button/ds6/button.css [hunk 3]
```diff
@@ -19,8 +20,7 @@ a.cta-btn {
 }
 a.cta-btn {
   background-color: #ffffff;
-  border-style: solid;
-  border-width: 1px;
+  border: 1px solid;
```
Comment (ianmcburnie) [inline dist/button/ds6/button.css]: This `border: 1px solid;` already appears to be defined in group selector above.

Comment (seangates) [reply dist/button/ds6/button.css]: Ah, let me grab that.

**Pull request merged**